app/process/watcher/crypto-transfer/crypto-transfer.go

```diff
@@ -157,32 +158,42 @@ func (ctw CryptoTransferWatcher) beginWatching(q *queue.Queue) {
 func (ctw CryptoTransferWatcher) processTransaction(tx transaction.HederaTransaction, q *queue.Queue) {
 	ctw.logger.Infof("New Transaction with ID: [%s]", tx.TransactionID)
 
-	var amount int64
+	var amount string
 	for _, tr := range tx.Transfers {
 		if tr.Account == ctw.accountID.String() {
-			amount = tr.Amount
+			amount = fmt.Sprint(tr.Amount)
 			break
 		}
 	}
 
+	wholeMemoCheck := regexp.MustCompile("^0x([A-Fa-f0-9]){40}-[1-9][0-9]+-[1-9][0-9]+$")
```

Comment:
I think that the regex should be:
`^0x([A-Fa-f0-9]){40}-[1-9][0-9]*-[1-9][0-9]*$` since using `+` we won't match f.e:
`0xe3846863823a23795c0195eb421a7eb7d0740eef-1000-2`